### DISCUSSION TITLE
How to add a BaseModel property type hint for a type created with `create_model`

### DISCUSSION THREAD

**Discussion by rafalstapinski** [pydantic/pydantic#4555] [Question]
👋 

I have an ORM library which uses Pydantic to bind database results to Python object instances. It uses Pydantic's `create_model` to create a factory for each table which in turn create these result object instances.

Here's a stripped down example showing the kind of logic the ORM does.

```python
from pydantic import create_model, BaseModel

class Table:
    
    def __new__(cls, **create_fields):
        return cls._create_model_factory()(**create_fields)
    
    def _create_model_factory(cls):
        
        # creates factory_model_kwargs based on fields defined on Table subclass
        # example of what this looks like for "MyTable" below:
        factory_model_kwargs = {'name': (str, ...), 'age': (int, ...)}

        return create_model(cls.__name__, **factory_model_kwargs)


class MyTable(Table):
    name: str
    age: int

my_table = MyTable(name="raf', age=10)
```

***

I would like to have some containers that include objects created from the `MyTable` - something akin to:

```python
class ContainerObject(BaseModel):
    my_table: MyTable

my_container = ContainerObject(my_table=my_table)
```

However, I'm unable to create `my_container`,  rightfully so, as `my_table` is not an _instance_ of `MyTable` but rather an instance of a model created by the `MyTable` factory. 

***

Does Pydantic support the ability to have a `BaseModel` member be a dynamic model? If so, what would the proper type hint for a `my_table` within `ContainerObject` be? Alternatively, is there some way for me to provide a helper function within the `MyTable` class that I can use to inject/type hint a convenience-validator for `ContainerObject`? Perhaps something like:

```python
class ContainerObject(BaseModel):
    my_table = MyTable.custom_PydanticField_for_MyTable()
```
 I am hoping to avoid having to create custom `@validator` functions at each container model for each dynamic model member if it's possible.

If it helps, <a href="https://github.com/rafalstapinski/porm/blob/593c0c4998bb015d695c3b1ea71fb8b293d0b76f/p3orm/table.py#L39-L69" >here</a> is the actual code within my ORM that creates these dynamic result models.

Any help would be greatly appreciated.

**Comment by rafalstapinski** [comment]
Hmm, I suppose a pretty naive solution might just be 

```python
class ContainerObject(BaseModel):
    my_table: object | MyTable
```

This skips the validation for the `my_table` member, unfortunately, but still provides the intellisense I need. 

Ideally, ContainerObject should still check that `my_table` has all the keys the dynamically defined `MyTable` factory instance would

**Reply by samuelcolvin** [reply]
You mean

```py
class ContainerObject(BaseModel):
    my_table: object | MyTable
```

**Reply by rafalstapinski** [reply]
Yep, type hint not assigning value.
